### PR TITLE
Feat: add gasLimit in operationV2

### PR DESCRIPTION
**Pull request by ChristanDaniel**
## Why

On the Polygon chain, we are experiencing occasional issues where the website estimates a gas limit below what is required, resulting in transaction errors.

## Changes

- Add gasLimit parameter to investment and withdrawal operations in OperationV2

### CODE REVIEW

File: src/services/operationV2.ts [hunk 1]
```diff
@@ -104,6 +111,12 @@ export default class operationV2 implements IOperations {
     return request
   }
 
+  gasMargin(estimateGas: string) {
+    const twentyPercentOfValue = 1.2
+    const calculateGas = Big(estimateGas).mul(twentyPercentOfValue).toFixed()
```
Comment (Tiago138) [inline src/services/operationV2.ts:116]: Is this right? You are doing a toFixed then parsing float and then rounding it.

File: src/services/operationV2.ts [hunk 2]
```diff
@@ -208,20 +221,39 @@ export default class operationV2 implements IOperations {
...
+      minTokenAmountOut: minPoolAmountOut
+    }
+    const options = {
+      from: userWalletAddress,
+      value: nativeValue
+    }
+
+    let gasLimit: BigNumberish | undefined
+    const polygonChainId = 137
+    if (parseFloat(this.poolInfo.chainId) === polygonChainId) {
```
Comment (Tiago138) [inline src/services/operationV2.ts:240]: chainId are int, maybe you should use parseInt.

File: src/services/operationV2.ts [hunk 3]
```diff
@@ -265,10 +297,14 @@ export default class operationV2 implements IOperations {
...
-    const jsonProvider = new JsonRpcProvider(networks[137].rpc)
+    const jsonProvider = new JsonRpcProvider(
+      networks[parseFloat(this.poolInfo.chainId)].rpc
+    )
+
+    const thirtyPercentOfValue = 1.3
     const feeData = await jsonProvider.getFeeData()
     const fee = Big(feeData?.maxFeePerGas?.toString() ?? '0').mul(
-      Big(estimateGas.toString() ?? '0')
+      Big(estimateGas.toString() ?? '0').mul(thirtyPercentOfValue)
```
Comment (Tiago138) [inline src/services/operationV2.ts:307]: if the type of estimateGas is number you don't need the toString

File: src/services/operationV2.ts [hunk 4]
```diff
@@ -473,24 +509,39 @@ export default class operationV2 implements IOperations {
       trasactionData
     )
 
+    let gasLimit: BigNumberish | undefined
+    const polygonChainId = 137
+    if (parseFloat(this.poolInfo.chainId) === polygonChainId) {
```
Comment (Tiago138) [inline src/services/operationV2.ts:514]: Maybe use parseInt

**Review by guilherme-correa-s — APPROVED** [review]

**Review by Tiago138 — APPROVED** [review]

**Pull request merged**